src/spikeinterface/preprocessing/tests/test_zero_padding.py

```diff
@@ -34,5 +36,227 @@ def test_zero_padding():
+
+    # Check the partial retrieval of the original series at the start
+    partial_original_traces = recording.get_traces(
+        start_frame=start_frame_original_traces, end_frame=end_frame_original_traces
+    )
+    partial_orignal_traces_from_padded_traces = padded_traces[:-end_padding_to_retrieve, :]
+    assert np.allclose(partial_orignal_traces_from_padded_traces, partial_original_traces)
+
+    # Check that the end of the padded traces is padded with zeros
+    last_padding = padded_traces[end_frame_original_traces:, :]
+    expected_padding = np.zeros((end_padding_to_retrieve, num_channels))
+    assert np.allclose(last_padding, expected_padding)
+
+
+@pytest.mark.parametrize("padding_start, padding_end", [(5, 5), (0, 5), (5, 0)])
+def test_trace_padded_recording_retrieve_traces_with_partial_padding(recording, padding_start, padding_end):
+    num_samples = recording.get_num_samples()
+    num_channels = recording.get_num_channels()
+
+    padded_recording = TracePaddedRecording(
+        parent_recording=recording,
+        padding_start=padding_start,
+        padding_end=padding_end,
+    )
+
+    # Define how many frames are padded at the start and end (take into account they could be 0)
+    number_of_padded_frames_at_start = 2 if padding_start > 0 else 0
+    number_of_paded_frames_at_end = 3 if padding_end > 0 else 0
+
+    # Start before the padding starts and ensure the end includes padding by going
+    # further than the number of samples of the original recording
+    start_frame = padding_start - number_of_padded_frames_at_start
+    end_frame = padding_start + num_samples + number_of_paded_frames_at_end
+
+    # Retrieve the traces which should have partial padding at the start and end
+    padded_traces = padded_recording.get_traces(start_frame=start_frame, end_frame=end_frame)
+
+    # Find that there as many frames padded a the start as expected
+    padded_traces_start = padded_traces[:number_of_padded_frames_at_start, :]
+    expected_zeros = np.zeros((number_of_padded_frames_at_start, num_channels))
+    assert np.allclose(padded_traces_start, expected_zeros)
+
+    # Then from padding_start to the number of samples plus padding_start this should be the original traces
+    frame_where_original_trace_ends = number_of_padded_frames_at_start + num_samples
+    original_traces_from_padded_traces = padded_traces[
+        number_of_padded_frames_at_start:frame_where_original_trace_ends, :
+    ]
+    original_traces = recording.get_traces()
+    assert np.allclose(original_traces_from_padded_traces, original_traces)
+
+    # Find that there as many frames padded at the end as expected
+    padded_traces_end = padded_traces[frame_where_original_trace_ends:, :]
+    expected_zeros = np.zeros((number_of_paded_frames_at_end, num_channels))
+    assert np.allclose(padded_traces_end, expected_zeros)
+
+
 if __name__ == "__main__":
-    test_zero_padding()
+    test_zero_padding_channel()
```

Comment:
I have the feeling that this tests are really really long for a this small task, no ?

Reply:
Where are you coming from, @samuelgarcia? It is unusual for people to argue for less testing but the opinion does exist out there. 

How are you thinking about this?

I am going to say something obvious but maybe is useful for establishing context and starting the conversation:
I used TTD here and each of those tests uncovered errors on my original implementation that I had to improve to pass. If someone wants to change -for example for optimization process- the function in the future then they will have this safeguard so they can  optimize away without being concerned about introducing new bugs.

Reply:
héhé. I am from France and you know it! :)
Is there ant correlation ? maybe...

Reply:
Just to add something here, the torought thesting gave me the confidence for implementing your suggestion here https://github.com/SpikeInterface/spikeinterface/pull/1706#discussion_r1225883702 without me being concerned that something will broken.